Classes/Common/Document.php

```diff
@@ -667,54 +676,45 @@ protected function getRawTextFromXml($id)
-                        $rawTextXml = simplexml_load_string($file);
-                        // Reset entity loader setting.
-                        libxml_disable_entity_loader($previousValueOfEntityLoader);
-                        // Reset libxml's error logging.
-                        libxml_use_internal_errors($libxmlErrors);
-                        // Get the root element's name as text format.
-                        $textFormat = strtoupper($rawTextXml->getName());
+                    // Get full text file.
+                    $fileContent = GeneralUtility::getUrl($this->getFileLocation($this->physicalStructureInfo[$id]['files'][$fileGrpFulltext]));
+                    if ($fileContent !== false) {
+                        $textFormat = $this->getTextFormat($fileContent);
                     } else {
-                        Helper::devLog('Couln\'t load fulltext file for structure node @ID "' . $id . '"', DEVLOG_SEVERITY_WARNING);
-                        return $rawText;
+                        $this->logger->warning('Couldn\'t load full text file for structure node @ID "' . $id . '"');
+                        return $fullText;
                     }
                     break;
                 }
             }
         } else {
-            Helper::devLog('Invalid structure node @ID "' . $id . '"', DEVLOG_SEVERITY_WARNING);
-            return $rawText;
+            $this->logger->warning('Invalid structure node @ID "' . $id . '"');
+            return $fullText;
         }
         // Is this text format supported?
-        if (
-            !empty($rawTextXml)
-            && !empty($this->formats[$textFormat])
-        ) {
-            if (!empty($this->formats[$textFormat]['class'])) {
-                $class = $this->formats[$textFormat]['class'];
-                // Get the raw text from class.
-                if (
-                    class_exists($class)
-                    && ($obj = GeneralUtility::makeInstance($class)) instanceof FulltextInterface
-                ) {
-                    $rawText = $obj->getRawText($rawTextXml);
-                    $this->rawTextArray[$id] = $rawText;
-                } else {
-                    Helper::devLog('Invalid class/method "' . $class . '->getRawText()" for text format "' . $textFormat . '"', DEVLOG_SEVERITY_WARNING);
-                }
-            }
+        // This part actually differs from previous version of indexed OCR
+        if (!empty($fileContent) && !empty($this->formats[$textFormat])) {
+            $fullText = $fileContent;
         } else {
-            Helper::devLog('Unsupported text format "' . $textFormat . '" in physical node with @ID "' . $id . '"', DEVLOG_SEVERITY_WARNING);
+            $this->logger->warning('Unsupported text format "' . $textFormat . '" in physical node with @ID "' . $id . '"');
         }
-        return $rawText;
+        return $fullText;
+    }
+
+    private function getTextFormat($fileContent)
```

Comment:
Don't forget the method documentation.

Reply:
I'm not sure what is wrong with displaying of this part of the code in the Files changed view. There comment is missing, but in fact it is already added.